### PR TITLE
fix: normalize tagStr values

### PR DESCRIPTION
**Pull request by rebelchris**
Added normalize function for tag values.
Also added specific test case to ensure some edge-cases

### CODE REVIEW

File: src/entity/posts/utils.ts [hunk 1]
```diff
@@ -74,17 +74,22 @@ export const mergeKeywords = async (
   keywords?: string[],
 ): Promise<{ mergedKeywords: string[]; allowedKeywords: string[] }> => {
   if (keywords?.length) {
+    const cleanedKeywords = uniqueifyArray(
+      keywords
+        .map((keyword) => keyword.replace(/[^a-zA-Z0-9-_#.]/g, ''))
```
Comment (capJavert) [inline src/entity/posts/utils.ts:79]: It would be cool to extract this regex as const to avoid multiple instantiation and also name it something better to denote what it is trying to clean.

File: __tests__/workers/postUpdated.ts [hunk 2]
```diff
@@ -237,6 +237,47 @@ it('should update post and modify keywords', async () => {
   expect(postKeywords.length).toEqual(2);
 });
 
+it('should save keywords in cleaned versions', async () => {
```
Comment (capJavert) [inline __tests__/workers/postUpdated.ts:240]: maybe also `should clean X and Y` just so we better understand how are keywords supposed to be cleaned down the line.

**Review by capJavert — APPROVED** [review]
Looks good, more of a visibility comment.

**Review by sshanzel — APPROVED** [review]
Great work with the cleanups! 🚀

**Review by idoshamun — COMMENTED** [review]
Just noticed I didn't submit my PR! 🤦 
It's such an unexpected behavior, sometimes you need to submit and sometimes no

File: src/entity/posts/utils.ts [hunk 3]
```diff
@@ -74,17 +74,22 @@ export const mergeKeywords = async (
   keywords?: string[],
 ): Promise<{ mergedKeywords: string[]; allowedKeywords: string[] }> => {
   if (keywords?.length) {
+    const cleanedKeywords = uniqueifyArray(
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply src/entity/posts/utils.ts:77]: We already test the base logic seperate for keywords do those not suffice?
(Or maybe misreading your comment)

[the remark this answers is not in the document]
Comment (rebelchris) [reply src/entity/posts/utils.ts:77]: Alright makes sense, will add the feedback :)

**Comment by rebelchris** [comment]
@idoshamun / @capJavert I abstracted + TSdoc'd the functions and wrote specific tests for them.
Let me know if that covers enough in your opinions.

**Review by idoshamun — APPROVED** [review]
Thanks a lot for addressing the feedback! Looks great
